stage #2.5/module #4 REST Basics - Spring Web/REST Basics/content.md

```diff
@@ -0,0 +1,526 @@
+```
+http://api.example.com/device-management/managed-devices.xml  /*Do not use xml in the URI*/
+http://api.example.com/device-management/managed-devices 	    /*This is correct URI*/
+```
+- **Never use CRUD function names in:**<br>
+We should not use URIs to indicate a CRUD function. URIs should only be used to uniquely identify the resources and not any action upon them.
+We should use HTTP request methods to indicate which CRUD function is performed.<br> Examples:<br>
+```
+HTTP GET http://api.example.com/device-management/managed-devices           //Get all devices
+HTTP POST http://api.example.com/device-management/managed-devices         //Create new Device
+HTTP GET http://api.example.com/device-management/managed-devices/{id}     //Get device for given Id
+HTTP PUT http://api.example.com/device-management/managed-devices/{id}     //Update device for given Id
+HTTP DELETE http://api.example.com/device-management/managed-devices/{id}  //Delete device for given Id
+```
+- **Use query component to filter URI collection:**<br>
+Often, you will encounter requirements where you will need a collection of resources sorted, filtered, or limited based on some specific resource attribute.
+For this requirement, do not create new APIs – instead, enable sorting, filtering, and pagination capabilities in resource collection API and pass the input parameters as query parameters.<br> Examples:<br>
+```
+http://api.example.com/device-management/managed-devices
+http://api.example.com/device-management/managed-devices?region=USA
+http://api.example.com/device-management/managed-devices?region=USA&brand=XYZ
+http://api.example.com/device-management/managed-devices?region=USA&brand=XYZ&sort=installation-date
+```
+- **Consistent subdomain names should be used for your APIs:**<br>
+The top-level domain and first subdomain names (e.g., soccer.restapi.org) of an API should identify its service owner. The full domain name of an API should add a subdomain named api.
+For example:
+```
+http://api.soccer.restapi.org
+```
+- **Consistent subdomain names should be used for your client developer portal:**<br>
+Many REST APIs have an associated website, known as a developer portal.
+For example:
+```
+http://developer.soccer.restapi.org
+```
+
+### Define API operations in terms of HTTP methods and HTTP response status codes
+HTTP has defined few sets of methods which indicates the type of action to be performed on the resources.
+_The URL is a sentence, where resources are nouns and HTTP methods are verbs._
+The below table summarises the use of mostly-used HTTP methods and HTTP response status codes.
+| HTTP Method  | CRUD           | Collection Resource (e.g. /users)                                                                      | Single Resource (e.g. /users/123)                                                |
```

Comment:
Looks like table is broken
![Screenshot_9](https://user-images.githubusercontent.com/34266487/161974283-bbb3c33a-f7f4-4c14-874c-68eb307e1433.jpg)
.